compiler/back_end/cpp/testcode/text_format_test.cc

```diff
@@ -93,6 +93,89 @@ TEST(TextFormat, UpdateFromText) {
   EXPECT_EQ(view.b().Read(), 4);
 }
 
+TEST(TextFormat, JsonOutput) {
+  ::std::array<char, 57> values = {};
+  ::std::iota(values.begin(), values.end(), 0);
+
+  const auto view = MakeJsonTestStructView(&values);
+  EXPECT_EQ(
+      "{\"one_byte_enum\": \"ZERO\", \"seven_bit_uint\": 1, \"one_bit_flag\": "
+      "false, \"one_byte_uint\": 2, \"two_byte_uint\": 1027, "
+      "\"four_byte_uint\": 134678021, \"eight_byte_uint\": "
+      "1157159078456920585, \"uint8_array\": [17, 18, 19, 20, 21, 22, 23, 24, "
```

Comment:
So: a very "fun" thing about JSON is that all numeric values are 64-bit *floating-point* values.  The `"eight_byte_uint"` value will not be parsed correctly by many JSON parsers, depending how closely they hew to the JSON specification.

The typical workaround for this is to serial integers as base 10 ASCII strings, either:

*   All integers are serialized as strings (this is what Emboss does internally for the serialized IR)
*   Integers whose range is larger than +/-2^53 are serialized as strings (so any 8-byte integer would be serialized as a string, no matter what its current value)
*   Integers whose actual value is outside of [-2^(53),+2^(53)] are serialized as strings (so an 8-byte integer might be serialized as a string or a double, depending on its current value)